dbt/models/model/schema.yml

```diff
@@ -4,61 +4,192 @@ sources:
+                - year
+                - run_id
+                - parameter_name
+              meta:
+                description: parameter range should be unique by year and run_id
         tags:
           - load_auto
 
       - name: parameter_search
         description: '{{ doc("table_parameter_search") }}'
+        data_tests:
+          - unique_combination_of_columns:
+              name: model_parameter_search_unique_by_year_run_id_and_iteration
+              combination_of_columns:
+                - year
+                - run_id
+                - iteration
+              config:
+                  error_if: ">2136"
+              meta:
+                description: parameter search should be unique by year, run_id, and iteration
         tags:
           - load_auto
 
       - name: performance
         description: '{{ doc("table_performance") }}'
+        data_tests:
+          - unique_combination_of_columns:
+              name: model_performance_unique
+              combination_of_columns:
+                - year
+                - run_id
+                - stage
+                - triad_code
+                - geography_type
+                - geography_id
+                - class
+              meta:
+                description: performance should be unique by year, run_id, stage, triad_code, geography_type, geography_id, and class
         tags:
           - load_auto
 
       - name: performance_quantile
         description: '{{ doc("table_performance_quantile") }}'
+        data_tests:
+          - unique_combination_of_columns:
+              name: model_performance_quantile_unique
+              combination_of_columns:
+                - year
+                - run_id
+                - triad_code
+                - stage
+                - geography_type
+                - geography_id
+                - class
+                - num_quantile
+                - quantile
+              meta:
+                description: >
+                  performance quantile should be unique by year, run_id, stage, triad_code,
+                  geography_type, by_class, geography_id, num_quantile, and quantile
         tags:
           - load_auto
 
       - name: shap
         description: '{{ doc("table_shap") }}'
+        data_tests:
+          - unique_combination_of_columns:
+              name: model_shap_unique_by_year_run_id_meta_pin_meta_and_card_num
+              combination_of_columns:
+                - year
+                - run_id
+                - meta_pin
+                - meta_card_num
+              config:
+                  error_if: ">524"
+              meta:
+                description: shap should be unique by year, run_id, meta_pin, and meta_card_num
         tags:
           - load_auto
 
       - name: test_card
         description: '{{ doc("table_test_card") }}'
+        data_tests:
+          - unique_combination_of_columns:
+              name: model_test_card_unique
```

Comment:
This one has a super high error level. Is there a key that is missing?

Reply:
That's pretty bizarre. I don't _think_ you're missing any key columns here. It's possible that in the past we had separate test set data frames for the linear vs lgbm model results, but I don't see any column that would separate them. IMO as long as more recent model runs aren't adding to the error count it's fine.